### PR TITLE
Do not return image tag if no_icon var is set

### LINKED ISSUE #4926: user_with_realname_and_icon() function returning wrong result with no_icon set to true
Issue/Feature description
====================

user_with_realname_and_icon() is commonly used with no_icon: true property, despite the fact it's not triggering any change in output of that function.

Expected result
============
user_with_realname_and_icon(..., no_icon: true) will return result of the function without img tag with icon.

Further information
===============
* Function is defined in src/api/app/helpers/webui/webui_helper.rb

### PR DESCRIPTION
**Pull request by krauselukas**
The no_icon variable had no effect

Fixes #4926

### CODE REVIEW

File: src/api/app/helpers/webui/user_helper.rb [hunk 1]
```diff
@@ -67,7 +67,11 @@ def user_with_realname_and_icon(user, opts = {})
         printed_name = "#{realname} (#{user.login})"
       end
 
-      user_image_tag(user) + ' ' + link_to(printed_name, user_show_path(user))
+      if opts[:no_icon]
+        link_to(printed_name, user_show_path(user))
+      else
+        user_image_tag(user) + ' ' + link_to(printed_name, user_show_path(user))
```
Comment (dmarcoux) [inline src/api/app/helpers/webui/user_helper.rb]: I would interpolate this: `"#{user_image_tag(user)} #{link_to(printed_name, user_show_path(user))}"`

**Comment by krauselukas** [comment]
@dmarcoux i applied your suggestion :)

File: src/api/app/helpers/webui/user_helper.rb [hunk 2]
```diff
@@ -67,7 +67,11 @@ def user_with_realname_and_icon(user, opts = {})
         printed_name = "#{realname} (#{user.login})"
       end
 
-      user_image_tag(user) + ' ' + link_to(printed_name, user_show_path(user))
+      if opts[:no_icon]
+        link_to(printed_name, user_show_path(user))
+      else
+        "#{user_image_tag(user)} #{link_to(printed_name, user_show_path(user))}".html_safe
```
Comment (Ana06) [inline src/api/app/helpers/webui/user_helper.rb]: @dmarcoux 

> I would interpolate this: `"#{user_image_tag(user)} #{link_to(printed_name, user_show_path(user))}"`

I disagree here... `link_to` is not returning an String (in which case interpolation would make sense) but an HTML tag (for which rails is already calling `html_safe`). Converting it to String by using interpolation and calling `html_safe` again is IMO not an improvement.

Comment (krauselukas) [reply src/api/app/helpers/webui/user_helper.rb]: @Ana06 that makes sense, i'm going to revert to earlier state

**Comment by dmarcoux** [comment]
I'm fine with the concerns @Ana06 raised.

**Pull request merged**